bedrock/mozorg/views.py

```diff
@@ -215,3 +220,56 @@ def moss_view(request):
         template_name = 'mozorg/moss/index.html'
 
     return l10n_utils.render(request, template_name)
+
+
+def oauth_fxa(request):
+    """
+    Acts as an OAuth relier for Firefox Accounts. Currently specifically tuned to handle
+    the OAuth flow for the Firefox Concert Series (Q4 2018).
+
+    If additional OAuth flows are required in the future, please refactor this method.
+    """
+    # expected state should be in user's cookies
+    stateExpected = request.COOKIES.get('fxaOauthState', None)
+
+    # provided state passed back from FxA - these state values should match
+    stateProvided = request.GET.get('state', None)
+
+    # code must be present - is in redirect querystring from FxA
+    code = request.GET.get('code', None)
+
+    error = False
+    cookie_age = 86400  # 1 day
+
+    # ensure all the data we need is present and valid
+    if not (stateExpected and stateProvided and code):
+        print('data error')
+        error = True
+    elif stateExpected != stateProvided:
+        print('state mismatch error')
+        error = True
+    else:
+        token = get_fxa_oauth_token(code)
+
+        if not token:
+            print('token error')
+            error = True
+        else:
+            email = get_fxa_profile_email(token)
```

Comment:
Call to the `fxa_concert_rsvp` function will be in this area right?

Reply:
I've got the basket call commented out below (in the non `error` flow), but yes, basically. 😁